packages/gatsby-cli/src/reporter/ink/reporter.js

```diff
@@ -0,0 +1,149 @@
+import React from "react"
+import { Static, Box } from "ink"
+import { globalTracer } from "opentracing"
+import util from "util"
+import Activity, { calcElapsedTime } from "./components/activity"
+import { Message } from "./components/messages"
+
+const tracer = globalTracer()
+
+export default class GatsbyReporter extends React.Component {
+  verbose = process.env.gatsby_log_level === `verbose`
+  state = {
+    verbose: false,
+    messages: [],
+    activities: {},
+  }
+
+  createActivity = (name, activityArgs) => {
+    const { parentSpan } = activityArgs
+    const spanArgs = parentSpan ? { childOf: parentSpan } : {}
+    const span = tracer.startSpan(name, spanArgs)
+
+    return {
+      start: () => {
+        this.setState(state => {
+          return {
+            activities: {
+              ...state.activities,
+              [name]: {
+                status: ``,
+                startTime: process.hrtime(),
+              },
+            },
+          }
+        })
+      },
+      setStatus: status => {
+        this.setState(state => {
+          const activity = state.activities[name]
+
+          return {
+            activities: {
+              ...state.activities,
+              [name]: {
+                ...activity,
+                status: status,
+              },
+            },
+          }
+        })
+      },
+      end: () => {
+        span.finish()
+        const activity = this.state.activities[name]
+
+        let successText = `${name} - ${calcElapsedTime(activity.startTime)} s`
+        if (activity.status) {
+          successText += ` — ${activity.status}`
+        }
+
+        this.onSuccess(successText)
+
+        this.setState(
+          state => {
+            const activities = { ...state.activities }
+            delete activities[name]
+
+            return {
+              activities,
+            }
+          },
+          () => {}
```

Comment:
Do we need this?

Reply:
nope :) just some debugging left :p I removed it.